jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/CachingHttpContentFactory.java

```diff
@@ -0,0 +1,571 @@
+            _characterEncoding = httpContent.getCharacterEncoding();
+            _compressedFormats = httpContent.getPreCompressedContentFormats();
+            _mimeType = httpContent.getMimeType();
+            _contentLength = httpContent.getContentLength();
+            _lastModifiedInstant = httpContent.getLastModifiedInstant();
+            _lastModified = httpContent.getLastModified();
+
+            _lastAccessed.set(NanoTime.now());
+        }
+
+        @Override
+        public long getContentLengthValue()
+        {
+            return _contentLengthValue;
+        }
+
+        @Override
+        public ByteBuffer getByteBuffer()
+        {
+            return _buffer;
+        }
+
+        @Override
+        public long getLastAccessedNanos()
+        {
+            return _lastAccessed.get();
+        }
+
+        @Override
+        public void setLastAccessedNanos(long nanosTime)
+        {
+            _lastAccessed.set(nanosTime);
+        }
+
+        @Override
+        public String getKey()
+        {
+            return _cacheKey;
+        }
+
+        @Override
+        public void retain()
+        {
+            _referenceCount.incrementAndGet();
+        }
+
+        @Override
+        public void release()
+        {
+            if (_referenceCount.decrementAndGet() == 0)
+            {
+                if (_buffer != null)
+                    _byteBufferPool.release(_buffer);
+                super.release();
+            }
+        }
+
+        @Override
+        public Set<CompressedContentFormat> getPreCompressedContentFormats()
+        {
+            return _compressedFormats;
+        }
+
+        @Override
+        public HttpField getETag()
+        {
+            return _etagField;
+        }
+
+        @Override
+        public String getETagValue()
+        {
+            return _etagField == null ? null : _etagField.getValue();
+        }
+
+        @Override
+        public String getCharacterEncoding()
+        {
+            return _characterEncoding;
+        }
+
+        @Override
+        public MimeTypes.Type getMimeType()
+        {
+            return _mimeType;
+        }
+
+        @Override
+        public HttpField getContentLength()
+        {
+            return _contentLength;
+        }
+
+        @Override
+        public Instant getLastModifiedInstant()
+        {
+            return _lastModifiedInstant;
+        }
+
+        @Override
+        public HttpField getLastModified()
+        {
+            return _lastModified;
+        }
+
+        @Override
+        public String getLastModifiedValue()
+        {
+            return _lastModifiedValue;
+        }
+
+        @Override
+        public boolean isValid()
+        {
+            return true;
+        }
+    }
+
+    protected static class NotFoundHttpContent implements CachingHttpContent
+    {
+        private final AtomicLong _lastAccessed = new AtomicLong();
```

Comment:
Looks like a `volatile long` is all you need.

Reply:
done